### PR TITLE
Response

### PR DESCRIPTION
**Pull request by resource11**
Resubmitting assessment.

### CODE REVIEW

File: lib/assessment.js [hunk 1]
```diff
@@ -5,7 +5,18 @@
...
 
+  var result = 0;
+
+  // iterate over args length and compare value to result
+  for (var i = 0; i < args.length; i++) {
+    if (args[i] < result) {
+      result = args[i];
+    }
+  }
+  return result;
```
Comment (srhoulam) [inline lib/assessment.js:19]: This function will return zero if it is passed arguments that are all greater than zero. If you're going to initialize `result` to a value that is not one of the arguments, then you should start it at the largest possible value in JS in order for it to never be less than the arguments. In fact, this is how the built-in `Math.min` works: the initial result value is `Infinity`. Try calling `Math.min` without any arguments!

4/5

File: lib/assessment.js [hunk 2]
```diff
@@ -13,7 +24,16 @@ var min = function min() {
...
+  // get the arguments, assign to args
+  var args = arguments;
+
+  var result = 0;
 
+  // iterate over args length and compare value to result
+  for (var i = 0; i < args.length; i++) {
+    result += args[i];
+    }
+  return result;
```
Comment (srhoulam) [inline lib/assessment.js:36]: Great. 1

File: lib/assessment.js [hunk 3]
```diff
@@ -21,6 +41,16 @@ var sum = function sum() {
...
+ var result = [];
+
+  // iterate over the specified length
+  for (var i = 0; i < length; i++) {
+    // set each index value of the result array to defaultValue
+    result[i] = defaultValue;
+    }
+
+  // return the new array
+  return result;
```
Comment (srhoulam) [inline lib/assessment.js:53]: Excellent. 1

File: lib/assessment.js [hunk 4]
```diff
@@ -30,6 +60,19 @@ var newArray = function newArray(length, defaultValue) {
...
 
+ var result = [];
+
+  // iterate over the specified length
+  for (var i = 0; i < length; i++) {
+    // set each index value of the result of invoking the defaultsFunction
+
+    // remember to actually pass the index of the array *to* the function
+    // much more helpful when reading a question when not time-pressed to comprehend the question! :)
+    result[i] = defaultsFunction(i);
```
Comment (srhoulam) [inline lib/assessment.js:71]: Yes, exactly!

File: lib/assessment.js [hunk 5]
```diff
@@ -30,6 +60,19 @@ var newArray = function newArray(length, defaultValue) {
...
+    }
+
+  // return the new array
+  return result;
```
Comment (srhoulam) [inline lib/assessment.js:75]: Good. 1

File: lib/assessment.js [hunk 6]
```diff
@@ -38,8 +81,38 @@ var newArray2 = function newArray2(length, defaultsFunction) {
...
+
 var addMethod = function addMethod(propertyName, method, obj) {
 
+  // so I can write '!obj' and JS knows that to check if that parameter
+  // was passed in? Or does this work because we are intending 'obj'
+  // to be referencing an object?
+
+  // are we using the word 'obj' as a semantic name for an object because
+  // we're intending to use that parameter as an object? We could just
+  // as easily use 'mary' for 'obj' as long as we know intent, yes?
```
Comment (srhoulam) [inline lib/assessment.js:93]: Yes, we can name it anything. What we name a parameter has no bearing on what is actually passed in. We _choose_ to name things in ways that are helpful to our comprehension.

File: lib/assessment.js [hunk 7]
```diff
@@ -38,8 +81,38 @@ var newArray2 = function newArray2(length, defaultsFunction) {
...
+  if (!obj) {
+    obj = {};
+  }
+
+  // so in this case, by writing our function this way, if the propertyName doesn't exist,
+  // when this function is invoked, we create it by writing the below statement and assign the parameter named 'method' to the propertyName, correct? And 'method' is a semantically-named parameter, yes?
+
+  // Additionally, if the propertyName *already* exists, whatever propert value that *already* exists
+  // will be replaced with whatever argument (in this case, a method), we pass in when we invoke the function.
+  // Correct?
```
Comment (srhoulam) [inline lib/assessment.js:104]: Yep.

File: lib/assessment.js [hunk 8]
```diff
@@ -38,8 +81,38 @@ var newArray2 = function newArray2(length, defaultsFunction) {
...
+  // *****
+  // Hopefully I'm being more clear here. So far, it either seems like I'm either not explaining my reasoning
+  // and level of understanding of the problem clearly and am getting dinged for that...
+  // ...or I'm just not understanding the question. I'm hoping I'm being clear.
+
+  obj[propertyName]= method;
+
+
+  // return the object. Just the object, yes? Did I read the question correctly?
+  return obj;
```
Comment (srhoulam) [inline lib/assessment.js:115]: Yep. This is a flawless answer. 1

**Comment by srhoulam** [comment]
4.8/5

**Comment by resource11** [comment]
Yes!

> ^.^< 
> Sent from my katPhone
>

**Pull request closed without merging**